core/src/main/java/org/elasticsearch/action/bulk/BulkRequest.java

```diff
@@ -329,6 +331,8 @@ public BulkRequest add(BytesReference data, @Nullable String defaultIndex, @Null
                                 versionType = VersionType.fromString(parser.text());
                             } else if ("_retry_on_conflict".equals(currentFieldName) || "_retryOnConflict".equals(currentFieldName)) {
                                 retryOnConflict = parser.intValue();
+                            } else if ("fields".equals(currentFieldName)) {
+                                fields = Strings.splitStringByCommaToArray(parser.text());
```

Comment:
Should we parse an actual list instead of a comma-delimited string list? I know the update request uses splitStringByCommaToArray but I think it is different since it gets the value from a query_string parameter while here you can provide actual json?